scripts/flow_track_correl.py

```diff
@@ -0,0 +1,414 @@
+
+    dr : int
+        Pixel offset increment between samples.
+
+    of0 : int
+        Central offset to start correlation from.
+
+    Returns
+    -------
+    np.ndarray
+        Cross-correlation array for the given frame pair.
+    """
+    a = preprocess_image(read_fits(file_a, hdu=hdu))
+    b = preprocess_image(read_fits(file_b, hdu=hdu))
+    acc = calculate_cross_correlation(a, b, np.arange(n_ofs), dr, of0)
+
+    return acc
+
+
+def accumulate_cross_correlation_across_frames(files: list, hdu: int, dt: int, sparsity: int, n_ofs: int, dr: int,
+                                               of0: int) -> np.ndarray:
+    """
+    Accumulate cross-correlation across frames in a list of FITS files.
+
+    Parameters
+    ----------
+    files : list
+        List of file paths to FITS files.
+
+    hdu : int
+        Index of the header data unit storing the data array.
+
+    dt : int
+        Frame offset (in frames) between time-offset image pairs.
+
+    sparsity : int
+        Interval between frames to skip when accumulating cross-correlation.
+
+    n_ofs : int
+        Number of pixel offsets to use in cross-correlation.
+
+    dr : int
+        Pixel offset increment between samples.
+
+    of0 : int
+        Central offset to start correlation from.
+
+    Returns
+    -------
+    np.ndarray
+        Accumulated cross-correlation array over all frames and offsets.
+    """
+    sample = read_fits(files[0], hdu=hdu)
+    acc = np.zeros((n_ofs, sample.shape[0], sample.shape[1]), dtype=float)
+
+    n = 0
+    for i in range(0, len(files) - dt, dt * sparsity):
+        print(f"Frame {i} vs frame {i + dt}")
+        acc += process_frame_pair(files[i], files[i + dt], hdu, n_ofs, dr, of0)
+        n += 1
+
+    acc /= n
+
+    return acc
+
+
+def compute_all_bands(acc: np.ndarray, ycen_band_rs: np.ndarray, chw: float, header, nlatbins: int,
+                      x_kps: np.ndarray):
+    """
+    Compute speed and sigma for all radial bands.
+
+    Parameters
+    ----------
+    acc : np.ndarray
+        Cross-correlation array accumulated across frames.
+
+    ycen_band_rs : np.ndarray
+        y-coordinates of band centers in solar radii.
+
+    chw : float
+        Half-width of each radial band in solar radii.
+
+    header : fits.header.Header
+        FITS header containing image information.
+
+    nlatbins : int
+        Number of angular bins for averaging.
+
+    x_kps : np.ndarray
+        Array mapping pixel offsets to speed in km/s.
+
+    Returns
+    -------
+    tuple
+        Tuple containing:
+        - np.ndarray : Average speed per angular bin for each radial band.
+        - np.ndarray : Sigma (standard deviation) of speed per angular bin for each radial band.
+    """
+    ylohi = calc_ylims(ycen_band_rs, chw, header)
+    # Determine spike location (index of the correlation peak) in the cross-correlation array
+    spike_location = np.where(x_kps < 0)[0].max() + 2
```

Comment:
Where does the +2 come from? I assume that'll never cause an out of bounds error for indexing by `spike_location`?